includes/component/plugin-tracking/class-tyche-plugin-tracking.php

```diff
@@ -0,0 +1,406 @@
+		 *
+		 * @var string $plugin_short_name
+		 */
+		private $plugin_short_name = '';
+
+		/**
+		 * Plugin Locale.
+		 *
+		 * @var string $plugin_locale
+		 */
+		private $plugin_locale = '';
+
+		/**
+		 * Blog Link.
+		 *
+		 * @var string $blog_link
+		 */
+		private $blog_link = '';
+
+		/**
+		 * Construct
+		 *
+		 * @since 1.1
+		 * @param array $options Options.
+		 */
+		public function __construct( $options ) {
+
+			if ( ! $this->init_vars( $options ) ) {
+				return;
+			}
+
+			add_action( $this->plugin_short_name . '_ts_tracker_send_event', array( &$this, 'send_tracking_data' ) );
+			add_action( $this->plugin_short_name . '_init_tracker', array( &$this, 'init_tracker' ) );
+			add_action( 'wp_ajax_' . $this->plugin_short_name . '_tracker_dismiss_notice', array( &$this, 'dismiss_notice' ) );
+			add_action( 'admin_notices', array( &$this, 'display_tracker_html_template' ) );
+			add_filter( 'cron_schedules', array( &$this, 'cron_schedule' ) );
+			add_action( 'admin_init', array( &$this, 'init_tracker' ) );
+			add_action( 'wp_ajax_ts_reset_tracking_setting', array( &$this, 'wcdn_reset_tracker_setting' ) );
+			$this->schedule_cron_job();
+		}
+
+		/**
+		 * Initialize variables from options array.
+		 *
+		 * @param array $options Options.
+		 *
+		 * @since 1.1
+		 */
+		public function init_vars( $options ) {
+
+			if ( ! is_array( $options ) ) {
+				return false;
+			}
+
+			if ( empty( array_diff( array( 'plugin_name', 'plugin_locale', 'plugin_short_name', 'version', 'blog_link' ), $options ) ) ) {
+				return false;
+			}
+
+			$this->plugin_name       = $options['plugin_name'];
+			$this->plugin_locale     = $options['plugin_locale'];
+			$this->plugin_short_name = $options['plugin_short_name'];
+			$this->version           = $options['version'];
+			$this->blog_link         = $options['blog_link'];
+
+			return true;
+		}
+
+		/**
+		 * Adds a weekly cron job schedule.
+		 *
+		 * @param array $schedules Schedules.
+		 */
+		public function cron_schedule( $schedules ) {
+			$schedules['once_in_week'] = array(
+				'interval' => 604800,  // one week in seconds.
+				'display'  => __( 'Once in a Week', $this->plugin_locale ), // phpcs:ignore
+			);
+
+			return $schedules;
+		}
+
+		/**
+		 * Cron Job Scheduler.
+		 */
+		public function schedule_cron_job() {
+			if ( ! wp_next_scheduled( $this->plugin_short_name . '_ts_tracker_send_event' ) ) {
+				wp_schedule_event( time() + 604800, 'once_in_week', $this->plugin_short_name . '_ts_tracker_send_event' );
+			}
+		}
+
+		/**
+		 * It will delete the tracking option from the database.
+		 *
+		 * @param string $plugin_short_name Plugin Short Name.
+		 */
+		public static function reset_tracker_setting( $plugin_short_name ) {
+			delete_option( $plugin_short_name . '_allow_tracking' );
+			delete_option( 'ts_tracker_last_send' );
+		}
+
+		/**
+		 * It will delete the tracking option from the database.
+		 */
+		public static function wcdn_reset_tracker_setting() {
```

Comment:
@nidhitatosaniya Why have you made changes to this file and added code here?

Reply:
Bcz i added ajax on click of 'reset tracking usage' button.

Reply:
@nidhitatosaniya You are not to make any change in that file. The file should be exactly as it is as you have copied it from the repo. Any change to that file must be done via a PR, so please revert those changes you have made in the file.

Reply:
@handelce  moved into main plugin class file.